projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php

```diff
@@ -665,4 +704,46 @@ public function auto_config_woo_result_format() {
 
 		update_option( Jetpack_Search_Options::OPTION_PREFIX . 'result_format', Jetpack_Search_Options::RESULT_FORMAT_PRODUCT );
 	}
+
+	/**
+	 * Save sidebars_widgets option before it's migrated by WordPress
+	 *
+	 * @since 9.8.0
+	 *
+	 * @param array $old_sidebars_widgets The sidebars_widgets option value to be saved.
+	 */
+	public function save_old_sidebars_widgets( $old_sidebars_widgets = null ) {
+		$this->old_sidebars_widgets = ! is_null( $old_sidebars_widgets ) ? $old_sidebars_widgets : wp_get_sidebars_widgets();
+	}
+
+	/**
+	 * Clean WordPress auto-migrated sidebar widgets from instant search sidebar before saving option sidebars_widgets
+	 *
+	 * @since 9.8.0
+	 *
+	 * @param array $sidebars_widgets The sidebars_widgets option value to be filtered.
+	 * @return array The sidebars_widgets option value to be saved
+	 */
+	public function remove_wp_migrated_widgets( $sidebars_widgets ) {
+		// Hook the action only when it is a theme switch i.e. $this->old_sidebars_widgets is not empty.
+		if ( empty( $this->old_sidebars_widgets ) ) {
+			return $sidebars_widgets;
+		}
+
+		foreach ( $sidebars_widgets as $sidebar => $widgets ) {
+			if ( 0 === stripos( $sidebar, static::JETPACK_INSTANT_SEARCH_SIDEBAR ) ) {
+				if ( count( $sidebars_widgets[ static::JETPACK_INSTANT_SEARCH_SIDEBAR ] ) <= count( $this->old_sidebars_widgets[ static::JETPACK_INSTANT_SEARCH_SIDEBAR ] ) ) {
```

Comment:
When would the count of widgets in the current sidebar be less than the widgets in the old sidebar?

Reply:
```
// An 'equal' would work for the condition, but we want to cover 'less than' as well, just for extra insurance.
// If the new Jetpack sidebar has already got less widgets, no need to run the code following the condition.
```
Added comments in 4fcef02